platform/util/platform-util/src/main/java/org/codice/ddf/platform/util/TemporaryFileBackedOutputStream.java

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.util;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+import com.google.common.io.ByteSource;
+import com.google.common.io.FileBackedOutputStream;
+
+/**
+ * TemporaryFileBackedOutputStream is a wrapper around {@link FileBackedOutputStream}. This class
+ * calls {@link FileBackedOutputStream#close()} and {@link FileBackedOutputStream#reset()} when
+ * {@link #close()} is called. This ensures that the temporary backing file is deleted. The method
+ * {@link #asByteSource()} should not be called after {@link #close()} is called.
+ */
+public class TemporaryFileBackedOutputStream extends OutputStream {
+
+    private FileBackedOutputStream fileBackedOutputStream;
+
+    private boolean isClosed = false;
+
+    /**
+     * @param fileThreshold the number of bytes before the stream should switch to buffering to a file
+     */
+    public TemporaryFileBackedOutputStream(int fileThreshold) {
+        this.fileBackedOutputStream = new FileBackedOutputStream(fileThreshold);
+    }
+
+    @Override
+    public void write(int b) throws IOException {
+        fileBackedOutputStream.write(b);
+    }
+
+    @Override
+    public void write(byte[] b) throws IOException {
+        fileBackedOutputStream.write(b);
+    }
+
+    @Override
+    public void write(byte[] b, int off, int len) throws IOException {
+        fileBackedOutputStream.write(b, off, len);
+    }
+
+    @Override
+    public void close() throws IOException {
+        try {
+            fileBackedOutputStream.close();
+            fileBackedOutputStream.reset();
+        } finally {
+            isClosed = true;
+        }
+    }
+
+    @Override
+    public void flush() throws IOException {
+        fileBackedOutputStream.flush();
+    }
+
+    /**
+     * Returns a readable {@link ByteSource} view of the data that has been written to this stream.
+     * Must not be called after {@link #close()} is called, otherwise an {@link IllegalStateException}
+     * will be thrown.
+     *
+     * @return ByteSource of the data
+     */
+    public ByteSource asByteSource() {
```

Comment:
This is a change to the FBOS contract.  We may never call asByteSource() after calling close(), but it has the potential of breaking code that currently works.